### ISSUE TITLE
Chapter 4 Neural Networks and Learning background behind learning rule

### ISSUE THREAD

**Issue by ChJus** [JC-ProgJava/Building-Neural-Networks-From-Scratch#22]
Discuss how the learning rule was derived and why it works.

**Comment by ChJus** [comment]
Also mention gradient descent
